client/src/pages/Interview.js

```diff
@@ -140,6 +154,20 @@ const Interview = (props) => {
     };
   }, [history, interviewId, socket]);
 
+  useEffect(() => {
+    if (socket) {
+      socket.on("code", (code) => {
+        setCode(code);
+      });
+      socket.on("compile", (result) => {
+        setResults(result);
+      });
+      socket.on("language", (language) => {
+        setLanguage(language);
+      });
+    }
+  }, []);
```

Comment:
you'll want to turn these listeners off ^ `code` `compile` `language` or else if you go back to the main page and enter this Interview component again, it'll keep replicating these listeners